packages/grid/src/elements/splitter/PaneProvider.tsx

```diff
@@ -0,0 +1,260 @@
+
+      setRowsTrack(state => {
+        const oldValue = rowsTrack[id];
+        const stealFromIndex = rows[id] + indexTraversal;
+
+        const stealFromKey = rowArray[stealFromIndex];
+
+        const difference = oldValue - value;
+
+        const nextState = {
+          ...state
+        } as Record<string, number>;
+
+        nextState[id] = value;
+        nextState[stealFromKey] = rowsTrack[stealFromKey] + difference;
+
+        return nextState;
+      });
+    },
+    [layoutIndices, rowsTrack, setRowsTrack]
+  );
+
+  const setColumnValue = useCallback(
+    (isStart: boolean, id: string, value: number) => {
+      const { columns, columnArray } = layoutIndices;
+      const indexTraversal = isStart ? -1 : 1;
+
+      setColumnsTrack(state => {
+        const oldValue = columnsTrack[id];
+        const stealFromIndex = columns[id] + indexTraversal;
+
+        const stealFromKey = columnArray[stealFromIndex];
+
+        const difference = oldValue - value;
+
+        const nextState = {
+          ...state
+        } as Record<string, number>;
+
+        nextState[id] = value;
+        nextState[stealFromKey] = columnsTrack[stealFromKey] + difference;
+
+        return nextState;
+      });
+    },
+    [layoutIndices, columnsTrack, setColumnsTrack]
+  );
+
+  const getLayoutValue = useCallback(
+    (dimension: DIMENSIONS, key: string, units?: UNITS) => {
+      switch (units) {
+        case 'px':
+          return layoutStateInPixels[dimension][key];
+        case 'fr':
+        default:
+          return dimension === 'rows' ? rowsTrack[key] : columnsTrack[key];
+      }
+    },
+    [columnsTrack, rowsTrack, layoutStateInPixels]
+  );
+
+  const getGridTemplateColumns = useCallback(
+    (units?: UNITS) => {
+      const { columnArray } = layoutIndices;
+
+      switch (units) {
+        case 'px':
+          return columnArray.map(col => `${layoutStateInPixels.columns[col]}px`).join(' ');
+        case 'fr':
+        default:
+          return columnArray.map(col => `${columnsTrack[col]}fr`).join(' ');
+      }
+    },
+    [layoutIndices, columnsTrack, layoutStateInPixels]
+  );
+
+  const getGridTemplateRows = useCallback(
+    (units?: UNITS) => {
+      const { rowArray } = layoutIndices;
+
+      switch (units) {
+        case 'px':
+          return rowArray.map(row => `${layoutStateInPixels.rows[row]}px`).join(' ');
+        case 'fr':
+        default:
+          return rowArray.map(row => `${rowsTrack[row]}fr`).join(' ');
+      }
+    },
+    [layoutIndices, rowsTrack, layoutStateInPixels]
+  );
+
+  const splitterContext = useMemo(
+    () => ({
+      columnState,
+      rowState,
+      setRowValue,
+      setColumnValue,
+      getLayoutValue,
+      totalPanesHeight,
+      totalPanesWidth,
+      pixelsPerFr
+    }),
+    [
+      rowState,
+      columnState,
+      setRowValue,
+      setColumnValue,
+      getLayoutValue,
+      totalPanesHeight,
+      totalPanesWidth,
+      pixelsPerFr
+    ]
+  );
```

Comment:
Could we add [a `useImperativeHandle`](https://reactjs.org/docs/hooks-reference.html#useimperativehandle) and a `forwardRef` to allow consumers to set the layout?

Reply:
After talking offline, we will punt on this as much the same value add is provided via the render props for exposing methods.